dev-tools/oxlog/src/bin/oxlog.rs

```diff
@@ -124,5 +130,31 @@ fn main() -> Result<(), anyhow::Error> {
             }
             Ok(())
         }
+        Commands::Services => {
+            let zones = Zones::load()?;
+
+            // We want all logs that exist, anywhere, so we can find their
+            // service names.
+            let filter = Filter {
+                current: true,
+                archived: true,
+                extra: true,
+                show_empty: true,
+            };
+
+            // Collect a unique set of services, based on the logs in all zones
```

Comment:
This could potentially be really slow in production. In the original incarnation of `oxlog`, I didn't require passing a zone at all and would search across all of them. Unfortunately, each zone is on a separate disk, and some have lots of files. This resulted in really slow behavior, on the order of seconds to 10s of seconds. I could also see this potentially interfering in production workloads, and because of that I forced a specific zone to be passed when looking for logs. In practice this isn't a huge deal, as you typically know where you want to look I believe.

I'd prefer we stick to that pattern here as well and force the user to pass a specific zone, and then look for the file names in that zone.

While I sympathize with wanting to see everything, I'm worried about the cost of accidentally doing this repeatedly on a production system. I think over time, we can possibly make some of this logic async and try to run it concurrently. While making the command faster, that may actually make the load worse though. It's possible I'm being too conservative here, but a command that takes 10 seconds to complete is also no fun to work with.

Reply:
makes sense. I'll switch it taking a zone argument then. `oxlog zones | xargs -n1 oxlog services` would then do the trick for looking at all zones, if I really wanted to

Reply:
Thanks @faithanalog. Conditionally approved on that change. 

Thanks for making this addition. It's super helpful!